arenadata_tookit: replace using pg_partitions

red1452 opened the pull request:
There was getting data from pg_partitions at function `adb_collect_table_stats`. `pg_partitions` generates from a lot of joins (`pg_class`, `pg_namespace`, `pg_partition_rule` and etc). It takes a lot of time and resources (query plan of getting all data from `pg_partitions` contains ~60 rows).

`pg_partitions` was replaces to some joins, which are need for `adb_collect_table_stats`. Query plans have halved than query with `pg_partitions`.

New test `adb_collect_table_stat_test` was added. It checks, that function `adb_collect_table_stats` generates new partitions for `db_files_history` correctly.

<details>
<summary>Query plan of checking if partition for current month exists for "db_files_history" before:</summary>

```sql
postgres=# explain (costs off) SELECT 1 FROM pg_catalog.pg_partitions p
                                        WHERE
                                        p.schemaname = 'arenadata_toolkit'
                                        AND
                                        p.partitionschemaname = 'arenadata_toolkit'
                                        AND
                                        p.tablename = 'db_files_history'
                                        AND
                                        now() BETWEEN CAST(substring(partitionrangestart FROM '#"%#"::%' FOR '#') AS TIMESTAMP WITHOUT TIME ZONE)
                                        AND
                                        CAST(substring(partitionrangeend FROM '#"%#"::%' FOR '#') AS TIMESTAMP WITHOUT TIME ZONE)
                                        AND
                                        p.partitiontype = 'range'
                                        AND
                                        partitionstartinclusive = 't';
                                                                                                                                                                                                                                            
                                                 QUERY PLAN                                                                                                                                                                                 
                                                                                                             

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

-------------------------------------------------------------------------------------------------------------
 Subquery Scan on p1
   Filter: (CASE WHEN (p1.partitiontype = 'range'::text) THEN p1.partitionstartinclusive ELSE NULL::boolean END AND (p1.schemaname = 'arenadata_toolkit'::name) AND (p1.partitionschemaname = 'arenadata_toolkit'::name) AND (p1.partitionty
pe = 'range'::text) AND ('2023-09-28 23:27:19.839481+03'::timestamp with time zone >= ("substring"(p1.partitionrangestart, '#"%#"::%'::text, '#'::text))::timestamp without time zone) AND ('2023-09-28 23:27:19.839481+03'::timestamp with 
time zone <= ("substring"(p1.partitionrangeend, '#"%#"::%'::text, '#'::text))::timestamp without time zone))
   ->  WindowAgg
         Partition By: pp.oid, pp.parlevel, cl3.relname
         Order By: pr1.parisdefault, pr1.parruleord
         ->  Sort
               Sort Key: pp.oid, pp.parlevel, cl3.relname, pr1.parisdefault, pr1.parruleord
               ->  WindowAgg
                     Partition By: pp.oid, pp.parlevel, cl3.relname
                     Order By: pr1.parruleord
                     ->  Sort
                           Sort Key: pp.oid, pp.parlevel, cl3.relname, pr1.parruleord
                           ->  Hash Join
                                 Hash Cond: (pg_database.dattablespace = s.oid)
                                 ->  Nested Loop
                                       ->  Hash Join
                                             Hash Cond: (cl.relnamespace = n.oid)
                                             ->  Hash Right Join
                                                   Hash Cond: (cl3.oid = pr2.parchildrelid)
                                                   ->  Seq Scan on pg_class cl3
                                                   ->  Hash
                                                         ->  Hash Left Join
                                                               Hash Cond: (pr1.parparentrule = pr2.oid)
                                                               ->  Hash Left Join
                                                                     Hash Cond: (cl.reltablespace = sp.oid)
                                                                     ->  Hash Join
                                                                           Hash Cond: (pp.parrelid = cl.oid)
                                                                           ->  Hash Left Join
                                                                                 Hash Cond: (cl2.reltablespace = sp3.oid)
                                                                                 ->  Hash Join
                                                                                       Hash Cond: (cl2.relnamespace = n2.oid)
                                                                                       ->  Hash Join
                                                                                             Hash Cond: (pr1.paroid = pp.oid)
                                                                                             ->  Hash Join
                                                                                                   Hash Cond: (cl2.oid = pr1.parchildrelid)
                                                                                                   ->  Seq Scan on pg_class cl2
                                                                                                   ->  Hash
                                                                                                         ->  Seq Scan on pg_partition_rule pr1
                                                                                             ->  Hash
                                                                                                   ->  Seq Scan on pg_partition pp
                                                                                                         Filter: (NOT paristemplate)
                                                                                       ->  Hash
                                                                                             ->  Seq Scan on pg_namespace n2
                                                                                 ->  Hash
                                                                                       ->  Seq Scan on pg_tablespace sp3
                                                                           ->  Hash
                                                                                 ->  Seq Scan on pg_class cl
                                                                                       Filter: (relname = 'db_files_history'::name)
                                                                     ->  Hash
                                                                           ->  Seq Scan on pg_tablespace sp
                                                               ->  Hash
                                                                     ->  Seq Scan on pg_partition_rule pr2
                                             ->  Hash
                                                   ->  Seq Scan on pg_namespace n
                                       ->  Materialize
                                             ->  Seq Scan on pg_database
                                                   Filter: (datname = 'postgres'::name)
                                 ->  Hash
                                       ->  Seq Scan on pg_tablespace s
 Optimizer: Postgres query optimizer
(60 rows)
```

</details>

<details>
<summary>Query plan of checking if partition for current month exists for "db_files_history" after:</summary>

```sql
postgres=# explain (costs off) SELECT 1            
                FROM pg_partition pp
                JOIN pg_class pgc1 ON pp.parrelid = pgc1.oid
                JOIN pg_partition_rule pr1 ON pr1.paroid = pp.oid
                JOIN pg_class pgc2 ON pgc2.oid = pr1.parchildrelid
                JOIN pg_namespace pgn1 ON pgc1.relnamespace = pgn1.oid
                JOIN pg_namespace pgn2 ON pgc2.relnamespace = pgn2.oid
                WHERE
                        pp.paristemplate = false
                        AND
                        pgn1.nspname = 'arenadata_toolkit'
                        AND
                        pgn2.nspname = 'arenadata_toolkit'
                        AND
                        pgc1.relname = 'db_files_history'
                        AND
                        now() BETWEEN CAST(
                                substring(pg_get_expr(pr1.parrangestart, pr1.parchildrelid) FROM '#"%#"::%' FOR '#')
                                AS TIMESTAMP WITHOUT TIME ZONE)
                        AND
                        CAST(substring(pg_get_expr(pr1.parrangeend, pr1.parchildrelid) FROM '#"%#"::%' FOR '#')
                                AS TIMESTAMP WITHOUT TIME ZONE)
                        AND
                        pp.parkind = 'r'  --range
                        AND
                        pr1.parrangestartincl = 't';
                                                                                                                                                                                                                 QUERY PLAN                 
                                                                                                                                                                                                 
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Hash Join
   Hash Cond: (pgc2.relnamespace = pgn2.oid)
   ->  Hash Join
         Hash Cond: (pgc2.oid = pr1.parchildrelid)
         ->  Seq Scan on pg_class pgc2
         ->  Hash
               ->  Hash Join
                     Hash Cond: (pgc1.relnamespace = pgn1.oid)
                     ->  Hash Join
                           Hash Cond: (pp.oid = pr1.paroid)
                           ->  Hash Join
                                 Hash Cond: (pgc1.oid = pp.parrelid)
                                 ->  Seq Scan on pg_class pgc1
                                       Filter: (relname = 'db_files_history'::name)
                                 ->  Hash
                                       ->  Seq Scan on pg_partition pp
                                             Filter: ((NOT paristemplate) AND (parkind = 'r'::"char"))
                           ->  Hash
                                 ->  Seq Scan on pg_partition_rule pr1
                                       Filter: (parrangestartincl AND ('2023-10-09 13:08:41.108181+03'::timestamp with time zone >= ("substring"(pg_get_expr(parrangestart, parchildrelid), '#"%#"::%'::text, '#'::text))::timestamp without
 time zone) AND ('2023-10-09 13:08:41.108181+03'::timestamp with time zone <= ("substring"(pg_get_expr(parrangeend, parchildrelid), '#"%#"::%'::text, '#'::text))::timestamp without time zone))
                     ->  Hash
                           ->  Seq Scan on pg_namespace pgn1
                                 Filter: (nspname = 'arenadata_toolkit'::name)
   ->  Hash
         ->  Seq Scan on pg_namespace pgn2
               Filter: (nspname = 'arenadata_toolkit'::name)
 Optimizer: Postgres query optimizer
(27 rows)
```

</details>

<details>
<summary>Query plan of inserting data to "db_files_current" before:</summary>

```sql
postgres=# explain (costs off) INSERT INTO arenadata_toolkit.db_files_current
        SELECT
                f.oid,
                f.table_name,
                f.table_schema,
                f.type,
                f.storage,
                part.tablename AS table_parent_table,
                part.schemaname AS table_parent_schema,
                f.table_database,
                f.table_tablespace,
                f."content",
                f.segment_preferred_role,
                f.hostname,
                f.address,
                f.file,
                CURRENT_TIMESTAMP AS modifiedtime,
                f.file_size
        FROM arenadata_toolkit.__db_files_current AS f
        LEFT JOIN pg_partitions AS part
                ON (part.partitionschemaname = f.table_schema AND part.partitiontablename = f.table_name);
                                                                                                    QUERY PLAN                                                                                                     

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Insert on db_files_current
   ->  Redistribute Motion 3:3  (slice6; segments: 3)
         ->  Nested Loop
               ->  Broadcast Motion 1:3  (slice1)
                     ->  Seq Scan on pg_tablespace tbl
               ->  Materialize
                     ->  Hash Left Join
                           Hash Cond: (segfiles.datoid = d.oid)
                           ->  Hash Left Join
                                 Hash Cond: (segfiles.tablespace_oid = t.oid)
                                 ->  Hash Left Join
                                       Hash Cond: (segfiles.reloid = c.oid)
                                       ->  Hash Join
                                             Hash Cond: (segfiles.dbid = gpconf.dbid)
                                             ->  Function Scan on adb_get_relfilenodes segfiles
                                             ->  Hash
                                                   ->  Broadcast Motion 1:3  (slice2)
                                                         ->  Seq Scan on gp_segment_configuration gpconf
                                       ->  Hash
                                             ->  Broadcast Motion 1:3  (slice3)
                                                   ->  Hash Left Join
                                                         Hash Cond: ((n.nspname = p1.partitionschemaname) AND (c.relname = p1.partitiontablename))
                                                         ->  Hash Left Join
                                                               Hash Cond: (c.relnamespace = n.oid)
                                                               ->  Seq Scan on pg_class c
                                                               ->  Hash
                                                                     ->  Seq Scan on pg_namespace n
                                                         ->  Hash
                                                               ->  Subquery Scan on p1
                                                                     ->  WindowAgg
                                                                           Partition By: pp.oid, cl.relname, pp.parlevel, cl3.relname
                                                                           Order By: pr1.parisdefault, pr1.parruleord
                                                                           ->  Sort
                                                                                 Sort Key: pp.oid, cl.relname, pp.parlevel, cl3.relname, pr1.parisdefault, pr1.parruleord
                                                                                 ->  WindowAgg
                                                                                       Partition By: pp.oid, cl.relname, pp.parlevel, cl3.relname
                                                                                       Order By: pr1.parruleord
                                                                                       ->  Sort
                                                                                             Sort Key: pp.oid, cl.relname, pp.parlevel, cl3.relname, pr1.parruleord
                                                                                             ->  Hash Join
                                                                                                   Hash Cond: (pg_database.dattablespace = s.oid)
                                                                                                   ->  Nested Loop
                                                                                                         ->  Hash Left Join
                                                                                                               Hash Cond: (cl2.reltablespace = sp3.oid)
                                                                                                               ->  Hash Join
                                                                                                                     Hash Cond: (cl2.relnamespace = n2.oid)
                                                                                                                     ->  Hash Right Join
                                                                                                                           Hash Cond: (cl3.oid = pr2.parchildrelid)
                                                                                                                           ->  Seq Scan on pg_class cl3
                                                                                                                           ->  Hash
                                                                                                                                 ->  Hash Join
                                                                                                                                       Hash Cond: (cl2.oid = pr1.parchildrelid)
                                                                                                                                       ->  Seq Scan on pg_class cl2
                                                                                                                                       ->  Hash
                                                                                                                                             ->  Hash Left Join
                                                                                                                                                   Hash Cond: (pr1.parparentrule = pr2.oid)
                                                                                                                                                   ->  Hash Left Join
                                                                                                                                                         Hash Cond: (cl.reltablespace = sp.oid)
                                                                                                                                                         ->  Hash Join
                                                                                                                                                               Hash Cond: (cl.relnamespace = n_1.oid)
                                                                                                                                                               ->  Hash Join
                                                                                                                                                                     Hash Cond: (pp.oid = pr1.paroid)
                                                                                                                                                                     ->  Hash Join
                                                                                                                                                                           Hash Cond: (cl.oid = pp.parrelid)
                                                                                                                                                                           ->  Seq Scan on pg_class cl
                                                                                                                                                                           ->  Hash
                                                                                                                                                                                 ->  Seq Scan on pg_partition pp
                                                                                                                                                                                       Filter: (NOT paristemplate)
                                                                                                                                                                     ->  Hash
                                                                                                                                                                           ->  Seq Scan on pg_partition_rule pr1
                                                                                                                                                               ->  Hash
                                                                                                                                                                     ->  Seq Scan on pg_namespace n_1
                                                                                                                                                         ->  Hash
                                                                                                                                                               ->  Seq Scan on pg_tablespace sp
                                                                                                                                                   ->  Hash
                                                                                                                                                         ->  Seq Scan on pg_partition_rule pr2
                                                                                                                     ->  Hash
                                                                                                                           ->  Seq Scan on pg_namespace n2
                                                                                                               ->  Hash
                                                                                                                     ->  Seq Scan on pg_tablespace sp3
                                                                                                         ->  Materialize
                                                                                                               ->  Seq Scan on pg_database
                                                                                                                     Filter: (datname = 'postgres'::name)
                                                                                                   ->  Hash
                                                                                                         ->  Seq Scan on pg_tablespace s
                                 ->  Hash
                                       ->  Broadcast Motion 1:3  (slice4)
                                             ->  Seq Scan on pg_tablespace t
                           ->  Hash
                                 ->  Broadcast Motion 1:3  (slice5)
                                       ->  Seq Scan on pg_database d
 Optimizer: Postgres query optimizer
(92 rows)
```

</details>

<details>
<summary>Query plan of inserting data to "db_files_current" after:</summary>

```sql
postgres=# explain (costs off) INSERT INTO pg_temp.db_files_current
        SELECT
                f.oid,
                f.table_name,
                f.table_schema,
                f.type,
                f.storage,
                pgc.relname AS table_parent_table,
                pgn.nspname AS table_parent_schema,
                f.table_database,
                f.table_tablespace,
                f."content",
                f.segment_preferred_role,
                f.hostname,
                f.address,
                f.file,
                CURRENT_TIMESTAMP AS modifiedtime,
                f.file_size
        FROM arenadata_toolkit.__db_files_current AS f
        LEFT JOIN pg_partition_rule pr ON pr.parchildrelid = f.oid
        LEFT JOIN pg_partition pp ON pr.paroid = pp.oid
        LEFT JOIN pg_class pgc ON pgc.oid = pp.parrelid
        LEFT JOIN pg_namespace pgn ON pgc.relnamespace = pgn.oid;
                                                         QUERY PLAN                                                          
-----------------------------------------------------------------------------------------------------------------------------
 Insert on db_files_current
   ->  Redistribute Motion 3:3  (slice6; segments: 3)
         ->  Nested Loop
               ->  Broadcast Motion 1:3  (slice1)
                     ->  Seq Scan on pg_tablespace tbl
               ->  Materialize
                     ->  Hash Left Join
                           Hash Cond: (segfiles.datoid = d.oid)
                           ->  Hash Left Join
                                 Hash Cond: (segfiles.tablespace_oid = t.oid)
                                 ->  Hash Left Join
                                       Hash Cond: (segfiles.reloid = c.oid)
                                       ->  Hash Join
                                             Hash Cond: (segfiles.dbid = gpconf.dbid)
                                             ->  Function Scan on adb_get_relfilenodes segfiles
                                             ->  Hash
                                                   ->  Broadcast Motion 1:3  (slice2)
                                                         ->  Seq Scan on gp_segment_configuration gpconf
                                       ->  Hash
                                             ->  Broadcast Motion 1:3  (slice3)
                                                   ->  Hash Left Join
                                                         Hash Cond: (c.relnamespace = n.oid)
                                                         ->  Hash Left Join
                                                               Hash Cond: (c.oid = pr.parchildrelid)
                                                               ->  Seq Scan on pg_class c
                                                               ->  Hash
                                                                     ->  Hash Left Join
                                                                           Hash Cond: (pgc.relnamespace = pgn.oid)
                                                                           ->  Hash Right Join
                                                                                 Hash Cond: (pp.oid = pr.paroid)
                                                                                 ->  Hash Right Join
                                                                                       Hash Cond: (pgc.oid = pp.parrelid)
                                                                                       ->  Seq Scan on pg_class pgc
                                                                                       ->  Hash
                                                                                             ->  Seq Scan on pg_partition pp
                                                                                 ->  Hash
                                                                                       ->  Seq Scan on pg_partition_rule pr
                                                                           ->  Hash
                                                                                 ->  Seq Scan on pg_namespace pgn
                                                         ->  Hash
                                                               ->  Seq Scan on pg_namespace n
                                 ->  Hash
                                       ->  Broadcast Motion 1:3  (slice4)
                                             ->  Seq Scan on pg_tablespace t
                           ->  Hash
                                 ->  Broadcast Motion 1:3  (slice5)
                                       ->  Seq Scan on pg_database d
 Optimizer: Postgres query optimizer
(48 rows)
```

</details>

red1452 committed:
arenadata_tookit: replace using pg_partitions(#629)

There was getting data from pg_partitions at function 'adb_collect_table_stats'.
'pg_partitions' generates from a lot of joins ('pg_class', 'pg_namespace',
'pg_partition_rule' and etc). It takes a lot of time and resources (query plan
of getting all data from 'pg_partitions' contains ~60 rows).

'pg_partitions' was replaces to some joins, which are need for
'adb_collect_table_stats'. Query plans have halved than query with 'pg_partitions'.

New test 'adb_collect_table_stat_test' was added. It checks, that function
'adb_collect_table_stats' generates new partitions for 'db_files_history' correctly.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/55663

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/55677

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.0--1.1.sql:
```diff
@@ -49,3 +49,131 @@ JOIN pg_catalog.pg_class pgc ON (skew.skewoid = pgc.oid)
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
```
What should I do if I already installed version 1.1 with a previous version of this function?

red1452 replied:
Version 1.1 is auto-generated from "1.0" and 1.0--1.1" scripts:
```
DATA = \
       arenadata_toolkit--1.0.sql \
       arenadata_toolkit--1.0--1.1.sql

DATA_built = $(EXTENSION)--$(EXTENSION_VERSION).sql

$(DATA_built): $(DATA)
	cat $(DATA) > $(DATA_built)
```

RekGRpth replied:
> Version 1.1 is auto-generated from "1.0" and 1.0--1.1" scripts:

Yes, but before this patch, the `arenadata_toolkit--1.1.sql` file already contained the `adb_collect_table_stats` function (which got there from the `arenadata_toolkit--1.0.sql` file), which means that before the patch I already had this function in version 1.1 and it will not update!

RekGRpth replied:
Try installing version 1.1 before the patch. How will the patch update the function later?

red1452 replied:
Moved to version 1.2

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/56120

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/687208

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/687210

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/687216

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,129 @@
+			JOIN pg_class pgc2 ON pgc2.oid = pr1.parchildrelid
+			JOIN pg_namespace pgn1 ON pgc1.relnamespace = pgn1.oid
+			JOIN pg_namespace pgn2 ON pgc2.relnamespace = pgn2.oid
+			WHERE pp.paristemplate = false) p
```
I think the p sub-select is redundant.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/56243

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/691862

RekGRpth commented on gpcontrib/arenadata_toolkit/sql/adb_collect_table_stats_test.sql:
```diff
@@ -0,0 +1,58 @@
+CREATE EXTENSION arenadata_toolkit;
+SET search_path = arenadata_toolkit;
```
Are there any collisions between the temporary table `db_files_current` from the `arenadata_toolkit.adb_collect_table_stats` function and the `arenadata_toolkit.db_files_current` table?

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,125 @@
+				'p'||to_char(now(), 'YYYYMM'));
+	END IF;
+
+	CREATE TEMPORARY TABLE IF NOT EXISTS db_files_current
```
```suggestion
	CREATE TEMPORARY TABLE IF NOT EXISTS pg_temp.db_files_current
```

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,125 @@
+		Since this table is temporary and user can call this several times in a session
+		then we need to truncate it (an alternative will be to drop it).
+	 */
+	TRUNCATE TABLE db_files_current;
```
```suggestion
	TRUNCATE TABLE pg_temp.db_files_current;
```

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,125 @@
+	   then checkpointing interval (there will be no unlink peformed), than these unmapped
+	   files will still go to db_files_current and db_files_history as unmapped.
+	*/
+	INSERT INTO db_files_current
```
```suggestion
	INSERT INTO pg_temp.db_files_current
```

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,125 @@
+
+	INSERT INTO arenadata_toolkit.db_files_current
+	SELECT *
+	FROM db_files_current;
```
```suggestion
	FROM pg_temp.db_files_current;
```

red1452 committed:
arenadata_tooklit: change temp db_files_current to pg_temp.db_files_current (#629)

At "adb_collect_table_stats" temporary table "db_files_current" is used for collecting
data from view "arenadata_toolkit.__db_files_current". Schema "arenadata_toolkit" contains
not temporary table "db_files_current". If schema "arenadata_toolkit" is set by default
it may be get collison at table names.

Temporary table "db_files_current" was renamed to "pg_temp.db_files_current".

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/56920

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/716773

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/716771

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/716767

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/720028

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/720030

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57057

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/721925

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/721926

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/726848

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/726849

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,124 @@
+	IF NOT EXISTS (
+		SELECT 1
+		FROM pg_partition pp
+		JOIN pg_class pgc1 ON pp.parrelid = pgc1.oid
```
I suggest such names
```diff
diff --git a/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql b/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql
index c7cbf74d4a..bfae820af1 100644
--- a/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql
+++ b/gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql
@@ -11,19 +11,19 @@ BEGIN
 	IF NOT EXISTS (
 		SELECT 1
 		FROM pg_partition pp
-		JOIN pg_class pgc1 ON pp.parrelid = pgc1.oid
+		JOIN pg_class cl ON pp.parrelid = cl.oid
 		JOIN pg_partition_rule pr1 ON pr1.paroid = pp.oid
-		JOIN pg_class pgc2 ON pgc2.oid = pr1.parchildrelid
-		JOIN pg_namespace pgn1 ON pgc1.relnamespace = pgn1.oid
-		JOIN pg_namespace pgn2 ON pgc2.relnamespace = pgn2.oid
+		JOIN pg_class cl2 ON cl2.oid = pr1.parchildrelid
+		JOIN pg_namespace n ON cl.relnamespace = n.oid
+		JOIN pg_namespace n2 ON cl2.relnamespace = n2.oid
 		WHERE
 			pp.paristemplate = false
 			AND
-			pgn1.nspname = 'arenadata_toolkit'
+			n.nspname = 'arenadata_toolkit'
 			AND
-			pgn2.nspname = 'arenadata_toolkit'
+			n2.nspname = 'arenadata_toolkit'
 			AND
-			pgc1.relname = 'db_files_history'
+			cl.relname = 'db_files_history'
 			AND
 			now() BETWEEN CAST(
 				substring(pg_get_expr(pr1.parrangestart, pr1.parchildrelid) FROM '#"%#"::%' FOR '#')
```

RekGRpth commented on gpcontrib/arenadata_toolkit/arenadata_toolkit--1.1--1.2.sql:
```diff
@@ -0,0 +1,124 @@
+		f.file,
+		CURRENT_TIMESTAMP AS modifiedtime,
+		f.file_size
+	FROM arenadata_toolkit.__db_files_current AS f
```
Where conditions
```sql
part.partitionschemaname = f.table_schema AND part.partitiontablename = f.table_name
```
?

red1452 replied:
This condition is not needed because there is filter by oid (`pg_partition_rule pr ON pr.parchildrelid = f.oid`).

RekGRpth commented on gpcontrib/arenadata_toolkit/expected/adb_collect_table_stats_test.out:
```diff
@@ -0,0 +1,319 @@
+ part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1                | arenadata_toolkit | part_table         | arenadata_toolkit   |         0
+ part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1                | arenadata_toolkit | part_table         | arenadata_toolkit   |         0
+ part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1                | arenadata_toolkit | part_table         | arenadata_toolkit   |         0
+ part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1_5_prt_sub_prt1 | arenadata_toolkit | part_table         | arenadata_toolkit   |    622592
```
Why part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1_5_prt_sub_prt1 appears 4 times?

red1452 replied:
Because this table has 4 files at 3 segments:
```
 part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1_5_prt_sub_prt1 | arenadata_toolkit | part_table         | arenadata_toolkit   | /home/evgeniy/gpdb/adb-6.x-gpdb/gpAux/gpdemo/datadirs/dbfast2/demoDataDir1/base/12812/17638_fsm |     98304
 part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1_5_prt_sub_prt1 | arenadata_toolkit | part_table         | arenadata_toolkit   | /home/evgeniy/gpdb/adb-6.x-gpdb/gpAux/gpdemo/datadirs/dbfast1/demoDataDir0/base/12812/17638     |     32768
 part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1_5_prt_sub_prt1 | arenadata_toolkit | part_table         | arenadata_toolkit   | /home/evgeniy/gpdb/adb-6.x-gpdb/gpAux/gpdemo/datadirs/dbfast2/demoDataDir1/base/12812/17638     |    622592
 part_table_1_prt_1_2_prt_1_3_prt_1_4_prt_1_5_prt_sub_prt1 | arenadata_toolkit | part_table         | arenadata_toolkit   | /home/evgeniy/gpdb/adb-6.x-gpdb/gpAux/gpdemo/datadirs/dbfast3/demoDataDir2/base/12812/17638     |     32768
```

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57202

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/727430

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/727431

BenderArenadata commented:
Failed job Behave tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/727429

BenderArenadata commented:
Failed job Behave tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/729444

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57258

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/729462

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/729463

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/729460

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/731065

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/731066
